python/GafferSceneUI/SetFilterUI.py

```diff
@@ -94,3 +98,148 @@
 	}
 
 )
+
+##########################################################################
+# NodeGadget drop handler
+##########################################################################
+
+__DropMode = enum.Enum( "__DropMode", [ "None_", "Add", "Remove", "Replace" ] )
+
+__originalDragPointer = None
+
+def __setsPlug( node ) :
+
+	for plug in node.children( Gaffer.Plug ) :
+		if Gaffer.Metadata.value( plug, "ui:scene:acceptsSetNames" ) or Gaffer.Metadata.value( plug, "ui:scene:acceptsSetExpression" ) :
+			return plug
+
+	return None
+
+def __dropMode( nodeGadget, event ) :
+
+	if __setsPlug( nodeGadget.node() ) is None :
+		filter = None
+		if nodeGadget.node()["filter"].getInput() is not None :
+			filter = nodeGadget.node()["filter"].source().node()
+		if filter is None :
+			return __DropMode.Replace
+		elif not isinstance( filter, GafferScene.SetFilter ) :
+			return __DropMode.None_
+
+	if event.modifiers & event.Modifiers.Shift :
+		return __DropMode.Add
+	elif event.modifiers & event.Modifiers.Control :
+		return __DropMode.Remove
+	else :
+		return __DropMode.Replace
+
+def __expressionSets( setExpression ) :
+
+	if (
+		"in" in setExpression or
+		"containing" in setExpression or
+		"|" in setExpression or
+		"&" in setExpression or
+		"(" in setExpression
+	) :
+		return [ f"( {setExpression} )" ]
+
+	return setExpression.split( " " )
+
+def __dragEnter( nodeGadget, event ) :
+
+	if not isinstance( event.data, IECore.StringVectorData ) :
+		return False
+
+	if not len ( event.data ) :
+		return False
+
+	if __dropMode( nodeGadget, event ) == __DropMode.None_ :
+		return False
+
+	global __originalDragPointer
+	__originalDragPointer = GafferUI.Pointer.getCurrent()
+
+	return True
+
+def __dragLeave( nodeGadget, event ) :
+
+	global __originalDragPointer
+
+	GafferUI.Pointer.setCurrent( __originalDragPointer )
```

Comment:
Don't know if this makes any difference, but I'm testing after rebasing on the latest `main`, which includes my recent drag-location-to-graph-editor work.

Reply:
Maybe we need the same `if __originalDragPointer is None` check that we have in `__dragMove`, so we only do this stuff when _we're_ the ones who accepted the drag in the first place?

Reply:
Fixed in https://github.com/GafferHQ/gaffer/pull/5786/commits/728bf78460ab4ece412e19470b7139a62f40d4f5

We also need to do the same thing in `PathFilterUI`, otherwise it will handle the signal and reset the pointer to the default.